core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -13,146 +13,623 @@
     if (this.isViewportAtMobileWidth()) {
       await this.page.waitForSelector('.e2e-test-toast-message', {
         visible: true,
       });
       await this.page.waitForSelector('.e2e-test-toast-message', {
         hidden: true,
       });
-      await this.clickOn(mobileChangesDropdown);
+      await this.clickOn(publishButtonDropDown);
       await this.clickOn(mobilePublishButton);
     } else {
-      await this.page.waitForSelector(
-        `${publishExplorationButton}:not([disabled])`
-      );
-      await this.clickOn(publishExplorationButton);
-    }
-    await this.clickOn(explorationTitleInput);
-    await this.type(explorationTitleInput, `${explorationTitle}`);
-    await this.clickOn(explorationGoalInput);
-    await this.type(explorationGoalInput, `${explorationTitle}`);
-    await this.clickOn(explorationCategoryDropdown);
-    await this.clickOn('Algebra');
-    await this.clickOn(saveExplorationChangesButton);
-    await this.clickOn(explorationConfirmPublishButton);
-    await this.page.waitForSelector(explorationIdElement);
-    const explorationIdUrl = await this.page.$eval(
-      explorationIdElement,
-      element => (element as HTMLElement).innerText
-    );
-    const explorationId = explorationIdUrl.replace(/^.*\/explore\//, '');
+      await this.clickOn(desktopPublishButton);
+    }
+    await this.clickOn(confirmPublishButton);
+    await this.page.waitForSelector(closePublishedPopUpButton, {visible: true});
+
+    explorationUrlAfterPublished = await this.page.url();
+    const explorationId = explorationUrlAfterPublished
+      .replace(/^.*\/create\//, '')
+      .replace(/#\/.*/, '');
+
+    await this.clickOn(closePublishedPopUpButton);
+    await this.expectExplorationToBePublish();
 
-    await this.clickOn(closeShareModalButton);
     return explorationId;
   }
 
+  async expectExplorationToBeAccessibleWithTheUrl(
+    expression: string
+  ): Promise<void> {
+    if (expression === 'Yes') {
+      try {
+        await this.page.goto(explorationUrlAfterPublished);
+        showMessage('Exploration is accessible with the URL i.e Published');
+      } catch (error) {
+        throw new Error('The exploration is not public.');
+      }
+    } else {
+      try {
+        await this.page.goto(explorationUrlAfterPublished);
+        throw new Error('The exploration is still public.');
+      } catch (error) {
+        showMessage(
+          'The exploration is not accessible with the URL i.e Deleted'
+        );
+      }
+    }
+  }
+
   /**
-   * Function to dismiss welcome modal
+   * This function discards the current changes.
    */
-  async dismissWelcomeModal(): Promise<void> {
-    await this.page.waitForSelector(dismissWelcomeModalSelector, {
+  async discardCurrentChanges(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(publishButtonDropDown);
+      await this.clickOn(mobileDiscardButton);
+    } else {
+      await this.page.keyboard.press('Tab');
```

Comment:
I don't think this was addressed: https://github.com/oppia/oppia/pull/19780#discussion_r1565917510

Reply:
done